diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e8b237..4306560 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -55,7 +55,12 @@ find_package(Git QUIET) # if we don't find git or FindGit.cmake is not on the sy
 if(GIT_FOUND) # GIT revision
     include(GetGitRevisionDescription)
     get_git_short_revision(GIT_SHA1)
-    message(STATUS "GIT revision ${GIT_SHA1}")
+    if(GIT_SHA1 STREQUAL "GITDIR-")
+       message(STATUS "Not building WPEFramework from git")
+       set(GIT_SHA1 "no_git")
+    else()
+       message(STATUS "GIT revision ${GIT_SHA1}")
+    endif()
     set(VERSION_REVISION ${GIT_SHA1})
 endif()
 
