commit c2cb09d305c181b359607933f275c841fe2c0153
Author: Sander van der Maar <sandervandermaar@gmail.com>
Date:   Fri Aug 18 12:08:19 2017 +0200

    Lowers CMake version

diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb87fcb..0afac9d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.5)
+cmake_minimum_required(VERSION 3.4)
 
 project(WPEFramework)
 
